src/huggingface_hub/commands/tag.py

```diff
@@ -0,0 +1,212 @@
+# coding=utf-8
+# Copyright 2023-present, the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Contains commands to perform tag management with the CLI.
+
+Usage Examples:
+    - Create a tag:
+        $ huggingface-cli tag create my-model 1.0 --tag-message "First release"
+        $ huggingface-cli tag create my-model 1.0 --tag-message "First release" --revision develop
+        $ huggingface-cli tag create my-model 1.0 --tag-message "First release" --type dataset
+    - List all tags:
+        $ huggingface-cli tag list my-model
+    - Delete a tag:
+        $ huggingface-cli tag delete my-model 1.0
+"""
```

Comment:
Hi @bilgehanertan, thanks for the PR! I haven't thoroughly reviewed it but implementation-wise the structure looks good. Before moving forward, I'd like to suggest some changes in the CLI format. Here is how I'd envision it:

```sh
# Tag model
$ huggingface-cli tag username/my-model 1.0
# Tag model with message
$ huggingface-cli tag username/my-model 1.0 -m "First release"
$ huggingface-cli tag username/my-model 1.0 --message "First release"
# Tag model with revision
$ huggingface-cli tag username/my-model 1.0 --revision develop
# Tag dataset
$ huggingface-cli tag username/my-dataset 1.0 --repo-type dataset

# List all model tags
$ huggingface-cli tag -l username/my-model
$ huggingface-cli tag --list username/my-model
# List all dataset tags
$ huggingface-cli tag -l username/my-dataset --repo-type dataset

# Delete model tag
$ huggingface-cli tag -d username/my-model 1.0
$ huggingface-cli tag --delete username/my-model 1.0
``` 

Main changes are:
- removed `create`/`delete`/`list` commands in favor of `-l`/`--list` and `-d`/`--delete` to list and delete tags. By default, tag creation is expected.
- updated `--type` to `--repo-type` for consistency
- I think namespace of the repo should always be explicit. 
- updated `--tag-message` to `--message` or `-m`

With this format, we would be much closer of the [`git tag`](https://git-scm.com/docs/git-tag)  command line that is already well established. What do you think?

Reply:
@Wauplin, thanks for the feedback! 
I have adjusted it & tested it; it should look better now.